src/helpers/initialize_data.jl

```diff
@@ -0,0 +1,13 @@
+"""
+    recursive_copyto!(a,b)
+
+copy the values from `b` to `a` by recursively copying the contents.
+On the lowest level of `AbstractArray`s this falls back to `copyto!`.
+"""
+function recursive_copyto!(a::AbstractArray{T}, b::AbstractArray{T}) where {T<:AbstractArray}
```

Comment:
```suggestion
function recursive_copyto!(a::AbstractArray{T}, b::AbstractArray{U}) where {T<:AbstractArray,U<:AbstractArray}
```
This won't force both element types to be the same.